src/fsharp/service/SemanticClassificationKey.fs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.SourceCodeServices
+
+open System
+open System.IO
+open System.IO.MemoryMappedFiles
+open System.Reflection.Metadata
+open System.Runtime.Serialization.Formatters.Binary;
+
+open FSharp.NativeInterop
+open FSharp.Compiler.Text
+open FSharp.Compiler.Text.Pos
+open FSharp.Compiler.Text.Range
+
+#nowarn "9"
+
+[<Sealed>]
+type SemanticClassificationKeyStoreView(mmf: MemoryMappedFile, length) =
+    member _.ReadRange(reader: byref<BlobReader>) =
+        let startLine = reader.ReadInt32()
+        let startColumn = reader.ReadInt32()
+        let endLine = reader.ReadInt32()
+        let endColumn = reader.ReadInt32()
+        let fileIndex = reader.ReadInt32()
+
+        let posStart = mkPos startLine startColumn
+        let posEnd = mkPos endLine endColumn
+        mkFileIndexRange fileIndex posStart posEnd
+
+    member this.ForEach() =
+        use view = mmf.CreateViewAccessor(0L, length)
```

Comment:
When the `ForEach` call completes, `view` will be disposed of, which means when the sequence starts iterating the view would have been disposed of already.

I suggest adding a function parameter:
```fsharp
member this.ForEach(f: struct(range * SemanticClassificationType) -> unit)
```
That way if we won't be trying to read from a disposed view when the call ends.

Reply:
I didn't get this completely. I thought the `yield` keyword kind of converts it into a state machine and the `view` would remain valid till all the elements of the sequences are emitted.